hw1/02-palindrome.js

```diff
@@ -1,3 +1,46 @@
-const elem = document.querySelector('input');
+const inputElem = document.querySelector("input");
+const NOT_A_NUMBER = "Not a number.";
+const POSITIVES_ONLY = "Input cannot be less than 0.";
+const NOT_PALINDROME = "No. Try again";
+const IS_PALINDROME = "Yes! This number is a palindrome!";
 
-elem.addEventListener('input', handleInput);
+inputElem.addEventListener("input", handleInput);
+
+function inputVerify(input) {
+    const verifyResult = isNaN(input) === true || input.length === 0;
+    return verifyResult;
+}
+
+function dangerElem(message) {
+    const modElem = document.getElementById("input-result");
+    modElem.classList = "";
+    modElem.classList.add("text-danger");
+    modElem.textContent = message;
+    return;
+}
+
+function successElem(message) {
+    const modElem = document.getElementById("input-result");
+    modElem.classList = "";
+    modElem.classList.add("text-success");
+    modElem.textContent = message;
+    return;
+}
+
+function handleInput() {
+    const receivedInput = inputElem.value;
+    const inputReversed = Array.from(receivedInput).reverse().join("");
+
+    if (inputVerify(receivedInput)) {
+        dangerElem(NOT_A_NUMBER);
+        return;
+    } else if (Number(receivedInput) < 0) {
+        dangerElem(POSITIVES_ONLY);
+        return;
+    } else if (receivedInput !== inputReversed) {
+        dangerElem(NOT_PALINDROME);
+        return;
+    }
+
+    elemRes = successElem(IS_PALINDROME);
+}
```

Comment:
**Rule 19.5** mentioned ending a file with a single newline character. Again, I don't know how important this is like the rule of using double or single quotes.

Reply:
Interesting, I'll check my source on my home pc but on my laptop that new line is present. Even viewing the file directly on gh shows it's missing. Wonder what's going on there.

Reply:
@PhatCash  Looks like if you don't have one GH will alert with a big red circle, but if you do, it simply ignores it and doesn't put an alert. Looks like it just eats that last line. See Baylie's files to see what I'm talking about and compare her files EOF line vs mine.

Reply:
Ah. I see what you're referring to. I have not seen that before, so it's interesting.